pkg/kubelet/kuberuntime/kuberuntime_container_linux_test.go

```diff
@@ -286,19 +290,35 @@ func TestCalculateLinuxResources(t *testing.T) {
 			},
 		},
 		{
-			name:   "RequestZeroCPU",
-			cpuReq: generateResourceQuantity("0"),
-			cpuLim: generateResourceQuantity("2"),
-			memLim: generateResourceQuantity("0"),
+			name:               "RequestZeroCPU",
+			cpuReq:             generateResourceQuantity("0"),
+			cpuLim:             generateResourceQuantity("2"),
+			memLim:             generateResourceQuantity("0"),
+			cgroup2UnifiedMode: false,
 			expected: &runtimeapi.LinuxContainerResources{
 				CpuPeriod:          100000,
 				CpuQuota:           200000,
 				CpuShares:          2,
 				MemoryLimitInBytes: 0,
 			},
 		},
+		{
+			name:               "Request128MBLimit256MBCgroup2",
+			cpuReq:             generateResourceQuantity("1"),
+			cpuLim:             generateResourceQuantity("2"),
+			memLim:             generateResourceQuantity("128Mi"),
+			cgroup2UnifiedMode: true,
+			expected: &runtimeapi.LinuxContainerResources{
+				CpuPeriod:          100000,
+				CpuQuota:           200000,
+				CpuShares:          1024,
+				MemoryLimitInBytes: 134217728,
+				Unified:            map[string]string{"memory.oom.group": "1"},
+			},
+		},
```

Comment:
yeah I think https://github.com/kubernetes/kubernetes/pull/119329 is good, but I think it's missing this section. I think we can also have tests that we skip for v1, and I think this case is a good addition @brianpursley

Reply:
This is the same test case as `Request128MBLimit256MB` but with the cgroup2-specific map entry (memory.oom.group).

Are you thinking I should duplicate the test cases for CGroup V2 and conditionally skip either the CGroup 1 or CGroup 2 tests depending on whether the host system has CGroup 1/2?